operator/README.md

```diff
@@ -1,2 +1,92 @@
+| kind | string | `Deployment`|
+| metaData | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Standard objects metadata. |
+| spec | [DeploymentSpec](#deployment-spec) | Specification of the desired behavior of the deployment |
+
+### DeployemtSpec
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| image | ImageInfo | PMEM-CSI driver container image information | `intel/pmem-csi-driver:canary` |
+| logLevel | integer | PMEM-CSI driver logging level | 3 |
+| controller | CotrollerDriver | configuration of PMEM-CSI driver running as controller | |
+| node | NodeDriver | configuration of PMEM-CSI driver running at compute node | |
+
+### ControllerDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| provisionerImager | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-provisioner:v1.2.1` |
+| resources | [ResourceRequirements](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | describes the compute resource requirements for controller pod |
+
+### NodeDriver
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| deviceMode | string | Device management mode to use. Supports one of `lvm` or `direct` | `lvm`
+| registrarImage | ImageInfo | CSI provisioner container image information | `quay.io/k8scsi/csi-node-driver-registrar:v1.1.0` |
+| resources | [ResourceList](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#resourcerequirements-v1-core) | |
+
+### ImageInfo
+
+|Field | type | Description | Default value |
+|---|---|--|---|
+| registry | string | Docker image registry |
+| name | string | Container image name |
+| tag | string | Container image tag |
+| pullPolicy | string | Image pull policy. either one of `Always`, `Never`, `IfNotPresent` | `IfNotPresent` |
+
+## Pre-requisites
+
+### Build operator image
+
+Build PMEM-CSI operator image using the provided [Dockerfile](https://github.com/intel/pmem-csi/tree/devel/operator/build/Dockerfile) and push the image to local docker registry.
+
+```sh
+docker build --pull -t pmem-csi-operator:canary  --label revision=$(VERSION) -f operator/build/Dockerfile .
+```
+
+### Deploy PMEM-CSI Deployment CRD
+
+```sh
+kubectl create -f https://raw.githubusercontent.com/avalluri/pmem-CSI/operator/operator/deploy/crds/pmem-csi.intel.com_deployments_crd.yaml
+```
+> FIXME: **Should this be handled with in operator**?
+
+### Deploy PMEM-CSI operator
+
+```sh
+kubectl create -f https://raw.githubusercontent.com/avalluri/pmem-CSI/operator/operator/deploy/operator.yaml
+```
+
+## Usage
+
+### Create a PMEM-CSI driver Deployment
+
+```sh
+kubectl create -f https://raw.githubusercontent.com/avalluri/pmem-CSI/operator/operator/deploy/crds/pmem-csi.intel.com_v1alpha1_deployment_cr.yaml
+```
+
+Once the deployment resource installation successful, we can see all the driver pods in Running state:
```

Comment:
Long term we should add a "deployment.Status" field where the operator itself reports whether PMEM-CSI is running properly.

Reply:
Exactly, It needs some brainstorming time for deciding what this Status should hold some useful information,